packages/server/modules/acc/clients/autodesk.ts

```diff
@@ -0,0 +1,308 @@
+}
+
+type AutodeskIntegrationTokenData = {
+  access_token: string
+  token_type: string
+  expires_in: number
+}
+
+/**
+ * Fetch a valid token for server-side operations as our custom integration
+ */
+export const getToken = async (): Promise<AutodeskIntegrationTokenData> => {
+  const clientId = getAutodeskIntegrationClientId()
+  const clientSecret = getAutodeskIntegrationClientSecret()
+
+  const token = Buffer.from(`${clientId}:${clientSecret}`, 'utf8').toString('base64')
+
+  const response = await fetch(
+    'https://developer.api.autodesk.com/authentication/v2/token',
+    {
+      method: 'POST',
+      body: new URLSearchParams({
+        grant_type: 'client_credentials',
+        scope: 'data:read account:read viewables:read'
+      }),
+      headers: {
+        Authorization: `Basic ${token}`,
+        Accept: 'application/json',
+        'Content-Type': 'application/x-www-form-urlencoded'
+      }
+    }
+  )
+
+  const data = await response.json()
+
+  return z
+    .object({
+      access_token: z.string(),
+      token_type: z.string(),
+      expires_in: z.number()
+    })
+    .parse(data)
+}
+
+/**
+ * Get base Autodesk API endpoint for a given region.
+ * NOTE: This has been true for all endpoints used so far but should be validated for any new ones!
+ */
+const getRegionUrl = (region: AccRegion): string => {
+  switch (region) {
+    case AccRegions.EMEA:
+      return 'https://developer.api.autodesk.com/modelderivative/v2/regions/eu'
+    default:
+      return 'https://developer.api.autodesk.com/modelderivative/v2'
+  }
+}
+
+const getApiUrl = (path: string, region: AccRegion): string => {
+  return `${getRegionUrl(region)}${path}`
+}
+
+/**
+ * Encode a urn string in the modified url-safe base64 format expected by the Autodesk API
+ */
+const encodeUrn = (urn: string): string => {
+  return btoa(urn).replaceAll('+', '-').replaceAll('/', '_').replaceAll('=', '')
+}
+
+type AccWebhookConfig = {
+  // The (unencoded) ACC folder urn to subscribe to. Event will fire for all files in this folder.
+  rootProjectFolderUrn: string
+  // The Speckle endpoint to hit when the given event fires.
+  callbackUrl: string
+  // The ACC webhook event to subscribe to. You can register an event to a given callback url only once.
+  event: 'dm.version.added'
+  // The region where the request is executed
+  region: AccRegion
+}
+
+/**
+ * Register relevant webhook callbacks for integration with a given ACC project.
+ * @see https://aps.autodesk.com/en/docs/webhooks/v1/reference/http/webhooks/systems-system-events-event-hooks-POST/
+ * @returns null if webhook already exists
+ */
+export const tryRegisterAccWebhook = async (
+  webhook: AccWebhookConfig
+): Promise<string | null> => {
+  const { rootProjectFolderUrn, callbackUrl, event, region } = webhook
+
+  const tokenData = await getToken()
+
+  const response = await fetch(
+    `https://developer.api.autodesk.com/webhooks/v1/systems/data/events/${event}/hooks`,
```

Comment:
in the future most if not all of the communication w/ ACC should happen from the backend, abstracted behind a GQL schema - the FE should not be invoking third party API endpoints